core/src/HIP/Kokkos_HIP_ParallelFor_Range.hpp

```diff
@@ -77,15 +78,20 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::HIP> {
     const int block_size =
         Kokkos::Impl::hip_get_preferred_blocksize<DriverType, LaunchBounds>(
             m_policy.space().hip_device());
-    const dim3 block(1, block_size, 1);
-    const dim3 grid(
-        typename Policy::index_type((nwork + block.y - 1) / block.y), 1, 1);
 
     if (block_size == 0) {
       Kokkos::Impl::throw_runtime_exception(
           std::string("Kokkos::Impl::ParallelFor< HIP > could not find a "
                       "valid execution configuration."));
     }
+    const dim3 block(1, block_size, 1);
+    const int max_work_items =
+        m_policy.space().hip_device_prop().maxGridSize[0];
+    const auto n_blocks = std::min<typename Policy::index_type>(
```

Comment:
If I recall correctly, `member_type` and `index_type` of a rangepolicy are the same thing. I wonder if it'd be better to use `Member` here (and on line 75) as well.

Reply:
Yes, they are the same and I prefer `index_type` but left `member_type` where it was used before.